Gems/Prefab/PrefabDependencyViewer/Code/Source/MainWindow.cpp

```diff
@@ -0,0 +1,264 @@
+        {
+            double currRight = (widestLevelSize - 1.0 * (numNodesAtEachLevel[level] - 1) / 2) * stepRight;
+            for (int nodeNum = 0; nodeNum < numNodesAtEachLevel[level]; ++nodeNum)
+            {
+                Utils::Node* currNode = queue.front();
+                queue.pop();
+
+                AZ::Vector2 pos(currRight, currDepth);
+
+                DisplayNode(currNode, pos);
+
+                Utils::NodeSet currChildren = graph.GetChildren(currNode);
+                for (Utils::Node* currChild : currChildren)
+                {
+                    queue.push(currChild);
+                }
+                currRight += stepRight;
+            }
+
+            currDepth += stepDown;
+        }
+
+        // Queue should be empty once done with all levels.
+        AZ_Assert(queue.empty(), "Queue should be empty.");
+    }
+
+    void PrefabDependencyViewerWidget::DisplayNode(Utils::Node* node, AZ::Vector2 pos)
+    {
+        const char* nodeStyle = "";
+        const AZ::Entity* graphCanvasNode = nullptr;
+        ;
+
+        GraphCanvas::GraphCanvasRequestBus::BroadcastResult(
+            graphCanvasNode, &GraphCanvas::GraphCanvasRequests::CreateGeneralNodeAndActivate, nodeStyle);
+
+        AZ_Assert(graphCanvasNode, "Unable to create GraphCanvas Node");
+
+        AZ::EntityId nodeUiId = graphCanvasNode->GetId();
+        nodeToNodeUiId[node] = nodeUiId;
+
+        GraphCanvas::NodeTitleRequestBus::Event(nodeUiId, &GraphCanvas::NodeTitleRequests::SetTitle, node->GetMetaData().GetSource());
+
+        GraphCanvas::SceneRequestBus::Event(m_sceneId, &GraphCanvas::SceneRequests::AddNode, nodeUiId, pos, false);
+        GraphCanvas::SceneMemberUIRequestBus::Event(nodeUiId, &GraphCanvas::SceneMemberUIRequests::SetSelected, true);
+
+        // Add slot
+        GraphCanvas::SlotLayoutRequestBus::Event(
+            nodeUiId, &GraphCanvas::SlotLayoutRequests::ConfigureSlotGroup, GraphCanvas::SlotGroups::DataGroup,
+            GraphCanvas::SlotGroupConfiguration(1));
+
+        GraphCanvas::SlotId inputSlotId = CreateDataSlot(nodeUiId, "Input", "Parent",
+                                                        azrtti_typeid<AZ::EntityId>(),
+                                                        GraphCanvas::SlotGroups::DataGroup, true);
+
+        GraphCanvas::SlotId outputSlotId = CreateDataSlot(nodeUiId, "Output", "Child",
+                                                        azrtti_typeid<AZ::EntityId>(),
+                                                        GraphCanvas::SlotGroups::DataGroup, false);
+
+        /* GraphCanvas::SlotId inputSlotId =
+                CreateExecutionSlot(nodeUiId, "Input", "Child",
+                                            GraphCanvas::SlotGroups::ExecutionGroup, true);
+        GraphCanvas::SlotId outputSlotId = CreateExecutionSlot(nodeUiId, "Output", "Parent",
+                                            GraphCanvas::SlotGroups::ExecutionGroup, false);
+        */
```

Comment:
more commented out code, does it require this to stay?

Reply:
Resolved